### PR TITLE
migrate string-refs in user_list.jsx

### PR DESCRIPTION
**Pull request by VolatianaYuliana**
#### Summary
change this.refs into React.createRef

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/15621

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/5753

### CODE REVIEW

**Review by marianunez — APPROVED** [review]

**Review by josephbaylon — CHANGES_REQUESTED** [review]
Hi @VolatianaYuliana Thanks for your contribution !! Please add unit tests for `user_list`. Here's the guide to writing unit tests: https://developers.mattermost.com/contribute/webapp/unit-testing/

**Comment by VolatianaYuliana** [comment]
Hey @josephbaylon! 
Sure, I will do that

**Comment by Willyfrog** [comment]
@VolatianaYuliana let us know if we can help you somehow.

**Comment by VolatianaYuliana** [comment]
Hey @Willyfrog! Sure! 
I am still learning how to write the unit tests that's why it takes a bit of time.
I will let you know if I need help :) thanks

**Comment by VolatianaYuliana** [comment]
Hi @Willyfrog!
I added the unit test for user_list.jsx. 
I'm gonna do the e2e now.

File: components/user_list.jsx [hunk 1]
```diff
@@ -32,9 +32,14 @@ export default class UserList extends React.PureComponent {
         rowComponentType: UserListRow,
     }
 
+    constructor(props) {
+        super(props);
+        this.container = React.createRef();
```
Comment (Willyfrog) [inline components/user_list.jsx:37]: can we rename this to match what we are doing in other PRs
```suggestion
        this.containerRef = React.createRef();
```

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Looking good, just two minor request:
- change the name to add a Ref suffix, we have this in almost all other requests
- remove the eslint-enable/disable comments as they are not needed anymore with your change

File: components/user_list.jsx (hunk 1 above)
Comment (VolatianaYuliana) [reply components/user_list.jsx:37]: okay @Willyfrog

**Review by Willyfrog — APPROVED** [review]
LGTM! thanks for working in this issue!

**Review by josephbaylon — APPROVED** [review]
LGTM. Verified unit test added and verified user list still loads in app

**Pull request merged**